src/include/sof/lib/clk.h

```diff
@@ -30,12 +27,18 @@ struct clock_notify_data {
 struct freq_table {
 	uint32_t freq;
 	uint32_t ticks_per_msec;
-	uint32_t enc;
 };
 
-extern struct freq_table *cpu_freq;
+struct clock_info {
+	uint32_t freqs_num;
+	struct freq_table *freqs;
+	uint32_t default_freq_idx;
+	uint32_t notification_id;
+	uint32_t notification_mask;
+	int (*set_freq)(int clock, int freq_idx);
```

Comment:
For future, we could also add a get() here, can be added later though in another PR